File: VBotMauiApp/Platforms/Android/HangupReceiver.cs
using Android.Content;
using VBot.Phone.SDK.Maui;
using VBotMauiApp.Services;

namespace VBotMauiApp.Platforms.Android;

[BroadcastReceiver(Enabled = true, Exported = false)]
public class HangupReceiver : BroadcastReceiver
{
    public override async void OnReceive(Context? context, Intent? intent)
    {
        if (context == null) return;
        OngoingCallNotification.Cancel(context);

        var phoneService = Microsoft.Maui.Controls.Application.Current?.Handler?.MauiContext?.Services.GetService(typeof(IVBotPhoneService)) as IVBotPhoneService;
        if (phoneService != null)
        {
            await phoneService.HangupAsync();
        }
    }
}

File: android/vbot-android-wrapper/src/main/kotlin/com/vpmedia/vbot/wrapper/HangupReceiver.kt
package com.vpmedia.vbot.wrapper

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import androidx.annotation.Keep

@Keep
class HangupReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        VBotAndroidWrapper.getInstance(context).hangup(null)
    }
}
